source/fundamentals/crud/read-operations/sort.txt

```diff
@@ -4,135 +4,396 @@ Sort Results
+
+.. _sorts-crud-sort-example:
+
+The examples in this guide use a sample collection, ``sort_example``, that
+contains the following documents:
+  
+.. code-block:: json
+  
+   {"_id": 1, "letter": "c", "food": "coffee with milk"}
+   {"_id": 3, "letter": "a", "food": "maple syrup"}
+   {"_id": 4, "letter": "b", "food": "coffee with sugar"}
+   {"_id": 5, "letter": "a", "food": "milk and cookies"}
+   {"_id": 2, "letter": "a", "food": "donuts and coffee"}
+   {"_id": 6, "letter": "c", "food": "maple donut"}
+
+Methods For Sorting
+-------------------
+
+You can sort your results using the  ``sort()`` method of a ``FindIterable``
+instance or by using the  ``Aggregates.sort()`` static factory method in an
+aggregation pipeline. Both of these methods take objects that implement the
+``Bson`` interface and specify sort criteria as arguments. For information
+on the ``Bson`` interface, see our 
+:doc:`guide on Bson </fundamentals/data-formats/document-data-format-bson/>`.   
+
+.. _sorts-crud-find-iterable-example:
+
+You can use the ``sort()`` method of a ``FindIterable`` instance as follows:
+
+.. code-block:: java
+
+   // <MongoCollection setup code here>
+   
+   collection.find().sort(new Document("_id", 1)));
+ 
+You can use the ``Aggregates.sort()`` method as follows:
+   
+.. code-block:: java
+
+   import com.mongodb.client.model.Aggregates;
+
+   // <MongoCollection setup code here>
+   
+   collection.aggregate(Arrays.asList(Aggregates.sort(new Document("_id", 1))));
+
+The code snippets above both sort the documents in the
+:ref:`sort_example collection <sorts-crud-sort-example>`  
+in the following order: 
+
+.. code-block:: json
+
+   {"_id": 1, "letter": "c", "food": "coffee with milk"}
+   {"_id": 2, "letter": "a", "food": "donuts and coffee"}
+   {"_id": 3, "letter": "a", "food": "maple syrup"}
+   ...
+
+In the above code snippets, we specify our sort criteria by passing a
+``Document`` object (which implements the ``Bson`` interface) to our sort
+method. While this will run without errors, we recommend that you instead
+specify sort criteria through the ``Sorts`` builder class. 
+
+The following code snippet is equivalent to the
+:ref:`FindIterable example above <sorts-crud-find-iterable-example>`. 
+
+.. code-block:: java
+
+   import static com.mongodb.client.model.Sorts.ascending;
+
+   // <MongoCollection setup code here>
+   
+   collection.find().sort(ascending("_id"));
+
+The remainder of this guide will specify sort criteria using the ``Sorts`` class.
+
+For more information on the benefits of using builders, see our
+:doc:`guide on builders </fundamentals/builders/>`. For more information on the
+``Sorts`` builder class, see our
+:doc:`guide on the Sorts builder </fundamentals/builders/sort/>`.    
+
+For more information about the classes and interfaces in this section, see the
+following API documentation:
+
+- :java-docs:`FindIterable </apidocs/mongodb-driver-sync/com/mongodb/client/FindIterable.html>` 
```

Comment:
Suggestion: Instead of having a bulleted list, you can link the java-docs where the class/interface is mentioned.  This avoids having to put it again at the bottom.

Reply:
This style was recommended in by Chris Cho [here](https://github.com/mongodb/docs-java/pull/65#discussion_r598887796) and [here](https://mongodb.slack.com/archives/GETCHDLCB/p1616618748074300?thread_ts=1616614407.070400&cid=GETCHDLCB) (see the second to the last bullet). That is the background for including the bulleted list. Please let me know if you disagree with this position, and if you do we should have a team meeting to reach consensus on style in regards to this issue.